conbench/tests/entities/test_commit.py

```diff
@@ -40,15 +40,28 @@ def test_repository_to_url():
 def test_get_github_commit_none():
     repo = "https://github.com/apache/arrow"
     sha = "3decc46119d583df56c7c66c77cf2803441c4458"
+    branch = "some_user_or_org:some_branch"
+    pr_number = 123
 
-    assert get_github_commit(None, None) == {}
-    assert get_github_commit("", "") == {}
-    assert get_github_commit(repo, None) == {}
-    assert get_github_commit(None, sha) == {}
+    assert get_github_commit(None, None, None, None) == {}
+    assert get_github_commit("", "", "", "") == {}
+    assert get_github_commit(repo, None, None, None) == {}
+    assert get_github_commit(None, pr_number, None, None) == {}
+    assert get_github_commit(None, None, branch, None) == {}
+    assert get_github_commit(None, None, None, sha) == {}
 
 
 @pytest.mark.slow
-def test_get_github_commit():
+@pytest.mark.parametrize(
+    "branch",
+    [
+        # as if supplied manually
+        None,
```

Comment:
I'm not sure I follow the comment here being supplied manually? Wouldn't `None` be like not supplying it?

Reply:
Yeah I can reword this comment. Basically, our recommended way to provide github info is to manually provide the `commit`, `repository`, and optionally `pr_number`. If you go the recommended way, you'll never provide `branch`.

If you go the "advanced" way (scraping with `github_info()`), then `branch` will be the auto-scraped branch string, which is the next test case here.

Reply:
That's great, thanks!